src/sage/structure/parent.pyx

```diff
@@ -1438,6 +1438,40 @@ cdef class Parent(sage.structure.category_object.CategoryObject):
         Hom_kwds = {} if category is None else {'category': category}
         return self.Hom(codomain, **Hom_kwds)(im_gens, **kwds)
 
+    def End(self, category=None):
+        r"""
+        Return the endomorphism set ``End(self, category)``, which is equivalent to
+        ``Hom(self, self, category)``.
+
+        INPUT:
+
+        - ``category`` -- a category or ``None`` (default: ``None``)
+          If ``None``, the category of ``self`` is used.
+
+        OUTPUT:
+
+        The endomorphism set of ``self`` in the category ``category``.
+
+        .. SEEALSO:: :func:`~sage.categories.homset.End`, :func:`~sage.categories.homset.Hom`
+
+        EXAMPLES::
+
+            sage: R.<x,y> = PolynomialRing(QQ, 2)
+            sage: R.End()
+            Set of Homomorphisms from Multivariate Polynomial Ring in x, y over Rational Field to Rational Field
```

Comment:
This doesn't look right

Reply:
Ah yes. I will fix it after Build&Test CI is done